x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/types.ts

```diff
@@ -128,11 +132,17 @@ export interface AboutStepRiskScore {
   isMappingChecked: boolean;
 }
 
+/**
+ * add / update data source types to show XOR relationship between 'index' and 'dataViewId' fields
+ * Maybe something with io-ts?
+ */
 export interface DefineStepRule {
   anomalyThreshold: number;
   index: string[];
   machineLearningJobId: string[];
   queryBar: FieldValueQueryBar;
+  dataViewId?: string;
+  dataViewTitle?: string;
```

Comment:
Is this being used anywhere?

Reply:
yup here: 34dd77511498ad2411064f4f96274d804a35fbf8